File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, n):  # Added 'self' to define it as a method
        if n < 0:
            sign = -1
        else:
            sign = 1  # Check for negative numbers
        rev = int(str(abs(n))[::-1])  # Reverse the number as a string
        if rev > 2**31 - 1:
                return 0
        return sign * rev

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int remainder = 0;
        long reverse = 0;
        while(x!=0){
            remainder = x%10;
            reverse = reverse *10;
            reverse = reverse + remainder;
            x = x/10;
        }
        if (reverse<std::pow(-2,31) || reverse>std::pow(2,31)-1){
            return 0;
        }
    return reverse;
    }
};